server/routes/api/documents/documents.ts

```diff
@@ -934,12 +943,21 @@ router.post(
 
     authorize(user, "update", original);
 
+    if (collectionId) {
+      const collection = await Collection.scope({
+        method: ["withMembership", user.id],
+      }).findByPk(collectionId, { transaction });
+      authorize(user, "createDocument", collection);
+    } else {
+      authorize(user, "createDocument", user.team);
+    }
+
     const document = await Document.create(
       {
         editorVersion: original.editorVersion,
-        collectionId: original.collectionId,
+        collectionId,
         teamId: original.teamId,
```

Comment:
```suggestion
        teamId: user.teamId,
```

Can we change this while we're here, just safer.